src/item_icon.c

```diff
@@ -157,12 +158,40 @@ u8 AddCustomItemIconSprite(const struct SpriteTemplate *customSpriteTemplate, u1
     }
 }
 
+static const u32 *const sTMTypePalettes[] =
```

Comment:
I think those could be moved to `gTypesInfo`? Not sure so please anyone else chime in.

Reply:
I agree but at the same time it seems like `gTypesInfo` should be moved to a more generic file than `battle_main.c` as it's used for non battle related stuff also.

Reply:
That's a good idea. Plus, we can take the chance to move the weakness table there as well.